pkg/providers/instancetype/suite_test.go

```diff
@@ -348,6 +348,9 @@ var _ = Describe("InstanceType Provider", func() {
 		It("should not include confidential SKUs", func() {
 			Expect(instanceTypes).ShouldNot(ContainElement(WithTransform(getName, Equal("Standard_DC8s_v3"))))
 		})
+		It("should not include SKUs without compatible image", func() {
+			Expect(instanceTypes).ShouldNot(ContainElement(WithTransform(getName, Equal("Standard_D2as_v6"))))
+		})
```

Comment:
A followup would be to have a unit test where we set the options to useSIG and then these skus are included, but we can add additional tests then.